src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2002,6 +2002,20 @@ private BoundExpression BindIncrementOperator(CSharpSyntaxNode node, ExpressionS
                 hasErrors);
         }
 
+        private BoundExpression BindNotNullExpression(PostfixUnaryExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            var expr = BindValue(node.Operand, diagnostics, BindValueKind.RValue);
```

Comment:
`!` can be used in `ref` contexts (see `NotNullOperator_RefOut` test) although those cases are not handled yet.